.g8/textAreaPage/generated-test/controllers/$package$/$className$ControllerSpec.scala

```diff
@@ -0,0 +1,160 @@
+
+  override def beforeEach: Unit = {
+    super.beforeEach
+    reset(mockUserAnswersCacheConnector)
+  }
+
+  "$className$ Controller" - {
+
+
+    "must return OK and the correct view for a GET" in {
+
+      val application = applicationBuilder(userAnswers = Some(emptyUserAnswers)).build()
+
+      running(application) {
+        val request = FakeRequest(GET, getRoute)
+
+        val result = route(application, request).value
+
+        val view = application.injector.instanceOf[$className$View]
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(form, waypoints)(request, messages(application)).toString
+      }
+    }
+
+    "must populate the view correctly on a GET when the question has previously been answered" in {
+
+      val userAnswers = UserAnswers().set($className$Page, validValue).success.value
+
+      val application = applicationBuilder(userAnswers = Some(userAnswers)).build()
+
+      running(application) {
+        val request = FakeRequest(GET, getRoute)
+
+        val view = application.injector.instanceOf[$className$View]
+
+        val result = route(application, request).value
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(form.fill(Some(validValue)), waypoints)(request, messages(application)).toString
+      }
+    }
+
+    "must save the answer and redirect to the next page when valid data is submitted (non empty value)" in {
+      when(mockUserAnswersCacheConnector.save(any(), any(), any())(any(), any()))
+        .thenReturn(Future.successful(()))
+
+      val application =
+        applicationBuilder(userAnswers = Some(emptyUserAnswers), extraModules)
+          .build()
+
+      running(application) {
+        val request =
+          FakeRequest(POST, postRoute).withFormUrlEncodedBody(("value", "abcdef"))
+
+        val result = route(application, request).value
+        val updatedAnswers = emptyUserAnswers.set($className$Page, validValue).success.value
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual $className$Page.navigate(waypoints, emptyUserAnswers, updatedAnswers).url
+      }
+    }
+
+    "must save the answer and redirect to the next page when valid data is submitted (empty value)" in {
+      when(mockUserAnswersCacheConnector.save(any(), any(), any())(any(), any()))
+        .thenReturn(Future.successful(()))
+
+      val application =
+        applicationBuilder(userAnswers = Some(emptyUserAnswers), extraModules)
+          .build()
+
+      running(application) {
+        val request =
+          FakeRequest(POST, postRoute).withFormUrlEncodedBody(("value", ""))
+
+        val result = route(application, request).value
+        val updatedAnswers = emptyUserAnswers.set($className$Page, validValue).success.value
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual $className$Page.navigate(waypoints, emptyUserAnswers, updatedAnswers).url
```

Comment:
It is better to verify mock at the end 
`        verify(mockUserAnswersCacheConnector, times(1)).save(any(), any(), any())(any(), any())
`

Reply:
Good point. Sorted and pushed.